lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -93,8 +93,11 @@ def parse(document, options = {})
           attributes['id'] = primary_data['id'] if primary_data['id']
           relationships = primary_data['relationships'] || {}
 
-          filter_fields(attributes, options)
-          filter_fields(relationships, options)
+          attributes = transform_keys(attributes, options)
+          relationships = transform_keys(relationships, options)
```

Comment:
Note that you can currently explicitly give "translations" for every field, which would solve your issue. I'm not sure making it easy to diverge from the spec's recommendations is a good idea (I know ember does it, but that's not a reason).